x/twap/listeners_test.go

```diff
@@ -3,35 +3,93 @@ package twap_test
 import (
 	"time"
 
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
 	"github.com/osmosis-labs/osmosis/v11/x/twap/types"
 )
 
-// TestCreatePoolFlow tests that upon a pool being created,
-// we have made the correct store entries.
-func (s *TestSuite) TestCreateTwoAssetPoolFlow() {
-	poolId := s.PrepareUni2PoolWithAssets(defaultUniV2Coins[0], defaultUniV2Coins[1])
-
-	expectedTwap, err := types.NewTwapRecord(s.App.GAMMKeeper, s.Ctx, poolId, denom0, denom1)
-	s.Require().NoError(err)
-
-	twap, err := s.twapkeeper.GetMostRecentRecordStoreRepresentation(s.Ctx, poolId, denom0, denom1)
-	s.Require().NoError(err)
-	s.Require().Equal(expectedTwap, twap)
-
-	twap, err = s.twapkeeper.GetRecordAtOrBeforeTime(s.Ctx, poolId, s.Ctx.BlockTime(), denom0, denom1)
-	s.Require().NoError(err)
-	s.Require().Equal(expectedTwap, twap)
-}
-
-// TODO: Write test for create pool, swap, and EndBlock in same block, we use post-swap spot price in record.
-
-// Tests that after a swap, we are triggering internal tracking logic for a pool.
-func (s *TestSuite) TestSwapTriggeringTrackPoolId() {
-	poolId := s.PrepareUni2PoolWithAssets(defaultUniV2Coins[0], defaultUniV2Coins[1])
-	s.BeginNewBlock(false)
-	s.RunBasicSwap(poolId)
-
-	s.Require().Equal([]uint64{poolId}, s.twapkeeper.GetChangedPools(s.Ctx))
+// TestAfterPoolCreatedHook tests if internal tracking logic has been triggered correctly,
+// and the corerct state entries have been created upon pool creation.
+// This test includes test cases for swapping on the same block with pool creation.
+func (s *TestSuite) TestAfterPoolCreatedHook() {
+	two_asset_coin := defaultUniV2Coins
+	three_asset_coins := defaultThreeAssetCoins
+
+	tests := []struct {
+		name      string
+		poolCoins sdk.Coins
+		// if this field is set true, we swap in the same block with pool creation
+		runSwap bool
+	}{
+		{
+			"Uni2 Pool, no swap on pool creation block",
```

Comment:
nit: for anyone unfamiliar with what univ2 means
```suggestion
			"two asset pool, no swap on pool creation block",
```